src/lando/api/legacy/api/stacks.py

```diff
@@ -67,13 +60,13 @@ def get(phab: PhabricatorClient, revision_id: str):
     )
     revision = phab.single(revision, "data", none_when_empty=True)
     if revision is None:
-        return not_found_problem
+        raise Http404("Revision does not exist")
 
     nodes, edges = build_stack_graph(revision)
     try:
         stack_data = request_extended_revision_data(phab, list(nodes))
     except ValueError:
-        return not_found_problem
+        raise Http404("Revision does not exist")
```

Comment:
Interesting though that on this particular line, this text might be misleading. I don't think it necessarily makes sense to include a more verbose message, since it could potentially expose a vulnerability (e.g., someone ends up seeing the custom message for whatever reason) since this particular error here is usually triggered due to access restrictions.

It's also not very clear that this text is only for debug purposes, and I still am not sure if it adds any value. In any case, I think we should keep it the same as the other error, or remove the error from both.